src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1003,6 +1003,19 @@ private static void NarrowFourUtf16CharsToAsciiAndWriteToBuffer(ref byte outputB
                 Vector128<uint> vecNarrow = Sse2.PackUnsignedSaturate(vecWide, vecWide).AsUInt32();
                 Unsafe.WriteUnaligned<uint>(ref outputBuffer, Sse2.ConvertToUInt32(vecNarrow));
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                // Narrows a vector of words [ w0 w1 w2 w3 ] to a vector of bytes
+                // [ b0 b1 b2 b3 b0 b1 b2 b3 ], then writes 4 bytes (32 bits) to the destination.
+
+                Vector128<short> vecWide = Vector128.CreateScalarUnsafe(value).AsInt16();
+                Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(vecWide);
+                unsafe
+                {
+                    AdvSimd.StoreSelectedScalar((byte*)Unsafe.AsPointer(ref outputBuffer), lower, 0);
```

Comment:
This is not correct. 
Based on Sse2 code, you need to store lower 4 bytes to memory, not 1. To do this you need to reinterpret `lower.AsUInt32()` and store the 0-th element. Don't forget also to adjust the comment:
```c#
                // Narrows a vector of 8 words [ w0 w1 w2 w3 * * * * ] to a vector of 8 bytes
                // [ b0 b1 b2 b3 * * * * ], then writes 4 bytes (32 bits) to the destination.
```
since you are using `Vector128.CreateScalarUnsafe()` to create `vecWide` that doesn't guarantee what values are in the upper half of the resulting `Vector<128>`, hence I put `*` to denote this.

Reply:
Done.